File: src/ds-visuals/stack/application/inToPost/code/stack.c
#include<stdlib.h>
#include<stdarg.h>
#define MAX 10
//stacks using arrays.

typedef struct stk{
	char arr[MAX];
	int top;
} Stack;

void push(Stack* stack, char element){
	if(stack->top == MAX-1)
		printf("Error- stack full");
	else
		stack->arr[++(stack->top)] = element;
	return;	
}

char pop(Stack*stack){
	if(stack->top == -1)
		printf("Error- Stack empty");
	else
		return stack->arr[(stack->top)--];	
	return -1;
}

char peek(Stack*stack){
	return stack->arr[stack->top];
}

Stack createStack(int num_elements, ...){
	Stack stack;
	int i;
	va_list list;
	stack.top = -1;
    va_start(list, num_elements); 
    for (i = 1; i <= num_elements; i++) 
        push(&stack, (char)va_arg(list, int)); 
    va_end(list); 
    return stack;  
}

File: src/ds-visuals/stack/code/stack.py
class Stack(object):
	"""docstring for Stack"""
	def __init__(self):
		super(Stack, self).__init__()
		self.top = -1
		self.MAX = 10
		self.stack = [0]*self.MAX

	def push(self,element):
		if(self.top == self.MAX-1):
			print("Error- stack full")
		else:
			self.top += 1
			self.stack[self.top] = element
		return;

	def pop(self):
		if(self.top == -1):
			print("Error- stack empty")
		else:
			self.top -= 1
			return self.stack[self.top+1]
		return -1

	def display(self):
		print('stack: ', end="")
		for i in range(self.top+1):
			print(str(self.stack[i])+" ", end="")
		print()

	def createStack(self, *elements):
		for element in elements:
			self.push(element)


s = Stack()
s.createStack(100, 200, 300)
s.display()
print("element popped: "+str(s.pop()))
s.display()
s.push(400)
s.display()
